kythe/cxx/indexer/cxx/KytheGraphObserver.h

```diff
@@ -140,13 +140,15 @@ class KytheGraphObserver : public GraphObserver {
                               const MetadataSupports* meta_supports,
                               const llvm::IntrusiveRefCntPtr<IndexVFS>& vfs,
                               ProfilingCallback ReportProfileEventCallback,
-                              std::string build_config = "")
+                              std::string build_config = "",
+                              std::string default_corpus = "")
```

Comment:
This is awkward; repeated function parameters of the same type (particularly when defaulted) are error-prone as it's often hard to tell at the call site which argument belongs to which parameter.

What was wrong with the previous formulation with the external call to `set_default_corpus`?

Reply:
`RegisterBuiltins()` and `EmitMetaNodes()` are called from the constructor, so the external call to `set_default_corpus()` was happening too late.

I could revert back to externally calling `set_default_corpus()` and also move `EmitMetaNodes()` outside the constructor.

Reply:
Seems like as good a time as any to pull the optional arguments into a separate struct, then.

1) remove set_default_corpus if it can't have the intended effect 
2) change to:
```
struct KytheGraphObserverOptions {
  // The custom build_config, if any, to apply to anchor nodes.
  std::string build_config = "";
  // The default corpus to use for nodes which would otherwise have an empty corpus.
  std::string default_corpus = "";
};
class KytheGraphObserver ... {
 public:
  // Nested classes in C++ are obnoxious, so Clang won't like the default construction below
  // if `Options` is a nested class.
  using Options = KytheGraphObserverOptions;

  explicit KytheGraphObserver(..., const Options& options = {}) ...
};
```

Reply:
:+1: